levels/challenge-questions/events.js

```diff
@@ -5,59 +5,117 @@ const fixDoorWallDepthSorting = require("./events/fixDoorDepthSorting");
 
 const WORLD_STATE_KEY = "com.twilioquest.challenge-questions";
 
-const INITIAL_STATE = {};
-
-module.exports = function (event, world) {
-  const worldState = merge(INITIAL_STATE, world.getState(WORLD_STATE_KEY));
-
-  const lockingObjectives = [];
+const INITIAL_STATE = {
+  challengeMapsCompleted: 0,
+  playerHasTeleported: false,
+  playerCompletedObjective: false
+};
 
-  if (event.name === 'mapDidLoad')
-    worldState.currentMapName = event.mapName;
+function mapIsCompleted(world, lockingObjectives, ...ignoredObjectives) {
+  return lockingObjectives && lockingObjectives.length &&
+    lockingObjectives.every((objectiveName) => {
+      if (!ignoredObjectives.includes(objectiveName))
+        return world.isObjectiveCompleted(objectiveName);
+      
+      return true;
+    });
+}
 
-  if (event.name === 'conversationDidEnd' && event.npc.conversation === 'cedricDefault' && worldState.chosenMap) {
-    world.warp('challenge-questions', 'player_entry1', worldState.chosenMap);
-    delete worldState.chosenMap;
-  }
+function getMapEvent(mapName) {
+  const mapEvent = {};
 
-  switch(worldState.currentMapName) {
+  switch(mapName) {
     case 'default':
-      lockingObjectives.push(
-        'difference-max-min',
-        'remove-duplicate-characters',
-        'reverse-words',
-        'sum-array'
-      );
+      mapEvent.lockingObjectives = ['difference-max-min', 'remove-duplicate-characters', 'reverse-words', 'sum-array', 'ducktypium-helm'];
```

Comment:
lockingObjectives contained all the objectives belonging to an individual map, whereas ignoredObjectives encapsulated the objectives that weren't necessary for removing barriers (e.g. the door on the first map or the laser in the second).

I've refactored the code a bit to make things more concise. Also: are we married to 'lockingObjectives' or would it be okay if I changed it? I used it here because I saw it in another objective

Reply:
If `lockingObjectives` does not make sense in the context of this `events.js` handler feel free to use a different name!